e2e/pages/app/column-picker/column-picker.testpage.component.ts

```diff
@@ -44,10 +44,16 @@ export class ColumnPickerTestPageComponent {
         'Work Completed'
     ];
 
+    selectedChange: ReadonlyArray<string>;
+
     groups: ColumnPickerGroup[] = [
         { name: 'Metadata', expanded: true }
     ];
 
+    _selectedChange(change: any) {
+        this.selectedChange = change;
```

Comment:
Establishing a duplicate of `this.selected` to test this issue indicates that there is another issue with the column picker. It looks like the array bound to the `selected` input is being modified in place, which is why it gets the updated value while `selectedChange` returns an incorrect value. That should be fixed instead of making this change to the test.

Also, don't use `any`, and give your functions return types.

Reply:
Hi Donna, thanks for the feedback on this. I have reverted the changes to this test suite. But I am unsure as to how to test the `selectedChange` value without the above function and changes to the test HTML. 

`selectedChange` now returns the correct value expected but I tried writing a Karma test for this, however it doesn't seem possible to reorder the selection from it to trigger the `selectedChange`

Further help on this would be appreciated.

Thanks,
Jonathan

Reply:
The array bound to the `selected` input is being modified in place (it is meant to be readonly). Once that is fixed, no additional test coverage is needed. `[(selected)]="selected"` is equivalent to `[selected]="selected" (selectedChange)="selected = $event"`, so you already have `selectedChange` hooked up in this test.

Reply:
Thanks for that Donna, sorry was in training most of today. 
So from this recommendation the selected input cannot change and can only be ReadonlyArray and therefore there would need to be another variable to store the augmented and changed state of the selected items when it is reordered or items are removed and added?

Sorry if I am completely missing misunderstanding this one.

Reply:
Yes, if the array is to be modified you need to create a whole new array which is normally done with the spread operator.
```
this.selected = [...this.selected];
```